app/views/qa/show.html.erb

```diff
@@ -31,38 +30,60 @@
-            items: options
-          } %>
+          <div class="govuk-!-margin-top-8">
+            <%= render "govuk_publishing_components/components/checkboxes", {
+              name: "#{current_facet['facet']['key']}[]",
+              heading: current_facet["question"],
+              description: render("govuk_publishing_components/components/govspeak", {
+                content: current_facet["description"].html_safe
+              }),
+              hint_text: current_facet["hint_title"],
+              is_page_heading: true,
+              items: options
+            } %>
+          </div>
         <% elsif single_wrapped_question? %>
           <% checkboxes = render "govuk_publishing_components/components/checkboxes", {
             name: "#{current_facet['facet']['key']}[]",
-            heading: "Select all that apply",
+            heading: current_facet["question"],
             visually_hide_heading: true,
             items: options
           } %>
-          <%= render "govuk_publishing_components/components/radio", {
-            name: "#{current_facet['facet']['key']}-yesno",
-            items: [
-              {
-                value: "yes",
-                text: "Yes",
-                conditional: checkboxes
-              },
-              {
-                value: "no",
-                text: "No"
-              }
-            ]
-          } %>
+          <div class="govuk-!-margin-top-8">
+            <%= render "govuk_publishing_components/components/radio", {
+              name: "#{current_facet['facet']['key']}-yesno",
+              heading: current_facet["question"],
+              description: render("govuk_publishing_components/components/govspeak", {
+                content: current_facet["description"].html_safe
+              }),
+              hint_text: current_facet["hint_title"],
+              is_page_heading: true,
+              items: [
+                {
+                  value: "yes",
+                  text: "Yes",
+                  conditional: checkboxes
+                },
+                {
+                  value: "no",
+                  text: "No"
+                }
+              ]
+            } %>
+          </div>
         <% else %>
-          <%= render "govuk_publishing_components/components/radio", {
-            name: "#{current_facet['facet']['key']}[]",
-            items: options
-          } %>
+          <div class="govuk-!-margin-top-8">
+            <%= render "govuk_publishing_components/components/radio", {
+              name: "#{current_facet['facet']['key']}[]",
+              heading: current_facet["question"],
+              description: render("govuk_publishing_components/components/govspeak", {
+                content: current_facet["description"].html_safe
+              }),
+              hint_text: current_facet["hint_title"],
+              is_page_heading: true,
+              items: options
+            } %>
+          </div>
```

Comment:
Could we refactor so that `<%= render "govuk_publishing_components/components/radio"` isn't repeated?

Something like:

```html
<%
  items = options
  if single_wrapped_question? {
    items = [
      { value: "yes"...},
      { value: "no"...}
    ]
  }
%>
<div class="govuk-!-margin-top-8">
  <%= render "govuk_publishing_components/components/radio", {
    name: "#{current_facet['facet']['key']}[]",
    heading: current_facet["question"],
    description: render("govuk_publishing_components/components/govspeak", {
      content: current_facet["description"].html_safe
    }),
    hint_text: current_facet["hint_title"],
    is_page_heading: true,
    items: items,
  } %>
</div>
```

Reply:
I'd say this is nice to have. I'm not a fan of lots of logic in the views because they don't seem to be as well tested as a presenter.

Reply:
Agreed it's a conditional in a view, which is a bit "😬" but it's only moving a conditional around - rather than adding one - and simplifying the template in the process.

Reply:
I say leave it as it is, the whole file needs to be refactored. We can improve this when there is a need.